File: Irrganten rb/irrgarten/CombatElement.rb
require_relative 'Dice'

module Irrgarten
  class CombatElement
    #Métodos
    #Constructor
    def initialize(e,u)
      @effect=e
      @uses=u
    end

    protected def produce_effect
      if (@uses > 0) then @uses=@uses-1
        return @effect
      else
        return 0
      end
    end


    def discard
      return Dice.discard_element(@uses)
    end

    def to_s
      return "["+@power.to_s+", "+@uses.to_s+"]"
    end


  end
end

File: irrgarten java/src/irrgarten/CombatElement.java
package irrgarten;

/**
 *
 * @author laurazafra
 */
public abstract class CombatElement {

    private float effect;
    private int uses;

    public CombatElement(float effect, int uses) {
        this.effect = effect;
        this.uses = uses;
    }

    //generaliza attack y protect
    protected float produceEffect() {
        if (uses > 0) {
            --uses;
            return effect;
        } else {
            return 0;
        }
    }

    public boolean discard() {
        return Dice.discardElement(uses); //usa la clase Dice
    }

    @Override
    public String toString() {
        return "[" + effect + ", " + uses + "]";
    }
}
